tests/components/dashboard_grid.test.ts

```diff
@@ -79,4 +79,30 @@ describe("Grid component in dashboard mode", () => {
     await simulateClick(".o-filter-icon");
     expect(fixture.querySelectorAll(".o-filter-menu")).toHaveLength(1);
   });
+
+  test("Clicking on a filter icon correctly closes context menu", async () => {
```

Comment:
small detail: "context menu" tends to refer to the right click menus
```suggestion
  test("Clicking on a filter icon correctly closes the popover", async () => {
```